### PR TITLE
#159965443 Implement article rating feedback. Represent average rate using stars.

### PR DESCRIPTION
**Pull request by Mutuba**
## What does this PR do?
```
Implements feedback to enable an average rating to be represented using stars. Users can now view the popularity of an article in terms of stars.
```

## Description of Task to be completed?
```
Enable users to view the popularity of an article using stars. Separate average ratings from user's ratings.

```

## How should this be manually tested?

```

Clone the repo ah-leagueOfLegends-frontend. cd into ah-leagueOfLegends-frontend and run npm install. Sign up if you do not have an account and sign in if you have an account. On home page, click on read more. Rate an article by clicking on any number of stars.

```
## Any background context you want to provide?

```
Users could see the popularity of an article using numeric values.

```

## What are the relevant pivotal tracker stories?

[#159965443](https://www.pivotaltracker.com/story/show/159965443
## Screenshots if available to support PR.

screen shot 2018-10-29 at 14 33 11

### CODE REVIEW

**Review by Derrickkip — APPROVED** [review]

**Review by Quantum-35 — APPROVED** [review]

**Review by DerKip — APPROVED** [review]

File: src/components/Rating/Rating.js [hunk 1]
```diff
@@ -37,31 +36,22 @@ class ArticleRating extends Component {
...
+                  // eslint-disable-next-line
                   this.props.ratingReducer.average_rating !== undefined
-                    ? this.props.ratingReducer.average_rating.toFixed(1)
-                    : this.props.average_rating}
-                </span>
-              </span>
+                    ? this.props.ratingReducer.average_rating
+                    : 0
+                }
+                // changeRating={this.handleStarClick}
```
Comment (kevintumbo) [inline src/components/Rating/Rating.js]: remove commented call

File: src/components/Rating/Rating.js [hunk 2]
```diff
@@ -72,14 +62,15 @@ class ArticleRating extends Component {
 
 const mapStateToProps = state => ({ ratingReducer: state.ratingReducer });
 
-ArticleRating.propTypes = {
-  InitialRate: PropTypes.func.isRequired,
+Rating.propTypes = {
+  // InitialRate: PropTypes.func.isRequired,
```
Comment (kevintumbo) [inline src/components/Rating/Rating.js]: remove commented code

File: src/components/Rating/UserRating.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+import PropTypes from 'prop-types';
+import StarRatings from 'react-star-ratings';
+import 'materialize-css/dist/css/materialize.min.css';
+import M from 'materialize-css';
+import { rateArticle, InitialRate } from './actions/actions';
+
+class ArticleRating extends Component {
+  constructor() {
+    super();
+    this.handleStarClick = this.handleStarClick.bind(this);
```
Comment (kevintumbo) [inline src/components/Rating/UserRating.js]: if you are using an arrow function then there is no need for using bind.

File: src/components/Rating/UserRating.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+            centerContent: 'flex-end',
+            marginLeft: '0%',
+            marginTop: '10%',
+            marginBottom: '10%',
+          }}
+        >
+          <div className="col s12 m6">
+            <span>My Rating: </span>
+            {' '}
+            {/* {console.log('Current rate', this.props.ratingReducer)} */}
```
Comment (kevintumbo) [inline src/components/Rating/UserRating.js]: remove commented out code

File: src/containers/Articles/Article/largArticle.js [hunk 5]
```diff
@@ -96,8 +103,7 @@ MainArticle.propTypes = {
 };
 
 const mapStateToProps = state => ({ mainArticle: state.completeArticle });
-const mapDispatchToProps = dispatch => bindActionCreators({ fetchOneArticles: fetchArticles },
-  dispatch);
+const mapDispatchToProps = dispatch => bindActionCreators({ fetchOneArticles: fetchArticles }, dispatch);
```
[the remark this answers is not in the document]
Comment (kevintumbo) [reply src/containers/Articles/Article/largArticle.js]: fix this max length error

**Review by kevintumbo — CHANGES_REQUESTED** [review]

File: src/components/Rating/UserRating.js (hunk 3 above)
Comment (Mutuba) [reply src/components/Rating/UserRating.js]: Thank you for advice.

File: src/components/Rating/UserRating.js (hunk 4 above)
Comment (Mutuba) [reply src/components/Rating/UserRating.js]: done

File: src/components/Rating/Rating.js (hunk 2 above)
Comment (Mutuba) [reply src/components/Rating/Rating.js]: done

**Review by kevintumbo — APPROVED** [review]

**Pull request merged**